pkg/sources/registration.go

```diff
@@ -0,0 +1,195 @@
+	events          []config.EventType
+	mappedResources []string
+	mappedEvent     config.EventType
+}
+
+func (i Registration) handleEvent(ctx context.Context, resource string, target config.EventType, sourceRoutes []Route, fn eventHandler) {
+	switch target {
+	case config.CreateEvent:
+		i.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				sources := sourcesForObjNamespace(ctx, sourceRoutes, obj, i.log, i.mapper, i.dynamicCli)
+				i.log.Debugf("handle CreateEvent, resource: %s, sources: %+v", resource, sources)
+				if len(sources) > 0 {
+					fn(ctx, resource, sources, nil)(obj)
+				}
+			},
+		})
+	case config.DeleteEvent:
+		i.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
+				sources := sourcesForObjNamespace(ctx, sourceRoutes, obj, i.log, i.mapper, i.dynamicCli)
+				i.log.Debugf("handle DeleteEvent, resource: %s, sources: %+v", resource, sources)
+				if len(sources) > 0 {
+					fn(ctx, resource, sources, nil)(obj)
+				}
+			},
+		})
+	case config.UpdateEvent:
+		i.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				sources, diffs := qualifySourcesForUpdate(ctx, newObj, oldObj, sourceRoutes, i.log, i.mapper, i.dynamicCli)
+				i.log.Debugf("handle UpdateEvent, resource: %s, sources: %+v, diffs: %+v", resource, sources, diffs)
+				if len(sources) > 0 {
+					fn(ctx, resource, sources, diffs)(newObj)
+				}
+			},
+		})
+	}
+}
+
+func (i Registration) handleMapped(ctx context.Context, targetEvent config.EventType, routeTable map[string][]Entry, fn eventHandler) {
+	i.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			var eventObj coreV1.Event
+			err := utils.TransformIntoTypedObject(obj.(*unstructured.Unstructured), &eventObj)
+			if err != nil {
+				i.log.Errorf("Unable to transform object type: %v, into type: %v", reflect.TypeOf(obj), reflect.TypeOf(eventObj))
+			}
+			_, err = cache.MetaNamespaceKeyFunc(obj)
+			if err != nil {
+				i.log.Errorf("Failed to get MetaNamespaceKey from event resource")
+				return
+			}
+
+			// Find involved object type
+			gvr, err := utils.GetResourceFromKind(i.mapper, eventObj.InvolvedObject.GroupVersionKind())
+			if err != nil {
+				i.log.Errorf("Failed to get involved object: %v", err)
+				return
+			}
+
+			if !i.canHandleEvent(eventObj.Type) {
+				return
+			}
+
+			gvrToString := utils.GVRToString(gvr)
+			if !i.includesSrcResource(gvrToString) {
+				return
+			}
+
+			sourceRoutes := sourceRoutes(routeTable, gvrToString, targetEvent)
+			sources := sourcesForObjNamespace(ctx, sourceRoutes, obj, i.log, i.mapper, i.dynamicCli)
+			if len(sources) == 0 {
+				return
+			}
+			fn(ctx, gvrToString, sources, nil)(obj)
+		},
+	})
+}
+
+func (i Registration) canHandleEvent(target string) bool {
+	for _, e := range i.events {
+		if strings.ToLower(target) == e.String() {
```

Comment:
what about using `strings.EqualFold`? Here is an example https://go.dev/play/p/DdBgSd-P-Ja

Reply:
Done.